packages/valory/contracts/gnosis_safe/tests/test_contract.py

```diff
@@ -628,3 +685,224 @@ def test_verify_reverted(self, *_: Any) -> None:
+
+        sender = crypto_registry.make(
+            EthereumCrypto.identifier, private_key_path=ETHEREUM_KEY_PATH_1
+        )
+        assert sender.address == self.owners()[1]
+        receiver = crypto_registry.make(
+            EthereumCrypto.identifier, private_key_path=ETHEREUM_KEY_PATH_2
+        )
+        assert receiver.address == self.owners()[2]
+        fourth = crypto_registry.make(
+            EthereumCrypto.identifier, private_key_path=ETHEREUM_KEY_PATH_3
+        )
+        assert fourth.address == self.owners()[3]
+        cryptos = [self.deployer_crypto, sender, receiver, fourth]
+        tx_hash = self.contract.get_raw_safe_transaction_hash(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            to_address=self.contract_address,
+            value=value,
+            data=data,
+        )["tx_hash"]
+        b_tx_hash = binascii.unhexlify(cast(str, tx_hash)[2:])
+        signatures_by_owners = {
+            crypto.address: crypto.sign_message(b_tx_hash, is_deprecated_mode=True)[2:]
+            for crypto in cryptos
+        }
+        assert list(signatures_by_owners.keys()) == self.owners()
+
+        tx = self.contract.get_raw_safe_transaction(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            sender_address=sender.address,
+            owners=(self.deployer_crypto.address.lower(),),
+            to_address=self.contract_address,
+            value=value,
+            data=data,
+            signatures_by_owner={
+                self.deployer_crypto.address.lower(): signatures_by_owners[
+                    self.deployer_crypto.address
+                ]
+            },
+        )
+
+        assert all(
+            key
+            in [
+                "chainId",
+                "data",
+                "from",
+                "gas",
+                "maxFeePerGas",
+                "maxPriorityFeePerGas",
+                "nonce",
+                "to",
+                "value",
+            ]
+            for key in tx.keys()
+        ), "Missing key"
+
+        prev_block = cast(int, self.ledger_api.api.eth.get_block_number()) + 1
+        tx_signed = sender.sign_transaction(tx)
+        tx_hash = self.ledger_api.send_signed_transaction(tx_signed)
+
+        assert tx_hash is not None, "Tx hash is `None`"
+
+        verified = self.contract.verify_tx(
+            ledger_api=self.ledger_api,
+            contract_address=self.contract_address,
+            tx_hash=tx_hash,
+            owners=(self.deployer_crypto.address.lower(),),
+            to_address=self.contract_address,
+            value=value,
+            data=data,
+            signatures_by_owner={
+                self.deployer_crypto.address.lower(): signatures_by_owners[
+                    self.deployer_crypto.address
+                ]
+            },
+        )
```

Comment:
this is exact same code block as [here](https://github.com/valory-xyz/open-autonomy/pull/1469/files#diff-23ee345d2f557f9dd1cd468e759f8cfecb17f49ee813d84105dc1d3826029cdcR774-R787). Please refactor